### ISSUE TITLE
local build is OK, Github pages gives undefined error

### ISSUE THREAD

**Issue by kerner1000** [mmistakes/so-simple-theme#254]
I have a blog post. 
If I add an featured image like this:
```
image:
  feature: my-image.jpg
  credit: Este
  creditlink: https://de.wikipedia.org
```
The local build is successful, GitHub nevertheless gives me an unspecified error:

![gh-build-fail](https://user-images.githubusercontent.com/13817521/29779238-38f56648-8c12-11e7-9324-4b1385dfb95b.png)

Jekyll Version is 
```
$ bundle exec jekyll -v
jekyll 3.4.3
```

**Comment by aborowska** [comment]
I have exactly the same problem with my website: locally it works fine, but there's an unspecified error on GitHub. I have jekyll 3.2.1.

Btw. I tried all the "tricks" from 
[https://github.com/mmistakes/so-simple-theme/issues/250](url)
to no success...

**Comment by mmistakes** [comment]
You'll want to edit your [`Gemfile`](https://github.com/mmistakes/so-simple-theme/blob/master/Gemfile) to make sure you're using the GitHub Pages gem. Which basically means replacing `gem "jekyll", "3.5.2"` with `gem "github-pages", group: :jekyll_plugins`.

Run `bundle update`, then `bundle exec jekyll build`.

This will build your site using the exact same gems and environment as GH-Pages. It should give you a more detailed error than what the website does when you push commits.

For more info on setting this up locally check [GitHub's docs](https://help.github.com/articles/setting-up-your-github-pages-site-locally-with-jekyll/).

**Comment by mmistakes** [comment]
I believe the issue here was with various Jekyll plugins like jekyll-feed and jekyll-seo-tag which are expecting `image: path-to-image.jpg` or `image.path: path-to-image.jpg`.

I've since fixed this with the 3.0 theme release. If you're still on v2 you can add `path` to the image object and those Jekyll plugins shouldn't complain anymore. 

```yaml
image:
  path:
  feature: my-image.jpg
  credit: Este
  creditlink: https://de.wikipedia.org
```

**Note:** `path` is intentionally left blank.

If you've upgraded to version 3.0.1, tested the build locally with the GitHub Pages gem, and still have the issue feel... free to reopen with the exact error Jekyll produces.
